udpa/core/v1/resource_locator.proto

```diff
@@ -0,0 +1,97 @@
+syntax = "proto3";
+
+package udpa.core.v1;
+
+import "udpa/annotations/status.proto";
+import "udpa/core/v1/context_params.proto";
+
+import "validate/validate.proto";
+
+option java_outer_classname = "ResourceLocatorProto";
+option java_multiple_files = true;
+option java_package = "com.github.udpa.udpa.core.v1";
+
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// UDPA resource locators identify a UDPA resource name and instruct the
+// data-plane load balancer on how the resource may be located.
+//
+// Resource locators have a canonical udpa:// URI representation:
+//
+//   udpa://{authority}/{type_url}/{id/*}?{context_params}{#directive*}
+//
+// where context_params take the form of URI query parameters.
+//
+// Resource locators have a similar canonical http:// URI representation:
+//
+//   http://{authority}/{type_url}/{id/*}?{context_params}{#directive*}
+//
+// Resource locators also have a simplified file:// URI representation:
+//
+//   file:///{id/*}{#directive*}
+//
+message ResourceLocator {
+  enum Schema {
+    UDPA = 0;
+    HTTP = 1;
+    FILE = 2;
+  }
+
+  // URI schema.
+  Schema schema = 1 [(validate.rules).enum = {defined_only: true}];
+
+  // Opaque identifiers for the resource. These are effectively concatenated
+  // with ‘/’ to form the non-query param path as resource ID. This may end
+  // with ‘*’ for glob collection references.
+  repeated string id = 2 [(validate.rules).repeated = {min_items: 1}];
+
+  // Logical authority for resource (not necessarily transport network address).
+  // Authorities are opaque in the UDPA API, data-plane load balancers will map
+  // them to concrete network transports such as an xDS management server, e.g.
+  // via envoy.config.core.v3.ConfigSource.
+  string authority = 3 [(validate.rules).string = {min_len: 1}];
+
+  // Qualified resource type (as in type URL without types.googleapis.com/
+  // prefix).
+  string qualified_type = 4 [(validate.rules).string = {min_len: 1}];
```

Comment:
I think `type` is reserved IIRC in proto3. The main point is that this is fully qualified type, rather than a type URL, which has the `types.googleapis.com/` prefix.

Reply:
I don't think "qualified" implies "fully qualified".  When I see "qualified" by itself, I think that there's some condition on the object (e.g., "qualified immunity"), which doesn't seem to be what you mean here.

Brainstorming some possible better names:

- fully_qualified_type
- full_type
- resource_type

Reply:
Sure, if you think `resource_type` is clearer, sold.